testing/integration/logs_ingestion_test.go

```diff
@@ -0,0 +1,439 @@
+			"Failed to download artifact",
+			"Failed to initialize artifact",
+			"Failed to apply initial policy from on disk configuration",
+			"elastic-agent-client error: rpc error: code = Canceled desc = context canceled", // can happen on restart
+		})
+	})
+	t.Logf("errors: Got %d documents", len(docs.Hits.Hits))
+	for _, doc := range docs.Hits.Hits {
+		t.Logf("%#v", doc.Source)
+	}
+	require.Empty(t, docs.Hits.Hits)
+
+	// Stage 4: Make sure we have message confirming central management is running
+	t.Log("Making sure we have message confirming central management is running")
+	docs = findESDocs(t, func() (estools.Documents, error) {
+		return estools.FindMatchingLogLines(info.ESClient, info.Namespace,
+			"Parsed configuration and determined agent is managed by Fleet")
+	})
+	require.NotZero(t, len(docs.Hits.Hits))
+
+	// Stage 5: verify logs from the monitoring components are not sent to the output
+	t.Log("Check monitoring logs")
+	hostname, err := os.Hostname()
+	if err != nil {
+		t.Fatalf("could not get hostname to filter Agent: %s", err)
+	}
+
+	agentID, err := fleettools.GetAgentIDByHostname(info.KibanaClient, policy.ID, hostname)
+	require.NoError(t, err, "could not get Agent ID by hostname")
+	t.Logf("Agent ID: %q", agentID)
+
+	// We cannot search for `component.id` because at the moment of writing
+	// this field is not mapped. There is an issue for that:
+	// https://github.com/elastic/integrations/issues/6545
+	// TODO: use runtime fields while the above issue is not resolved.
+
+	docs = findESDocs(t, func() (estools.Documents, error) {
+		return estools.GetLogsForAgentID(info.ESClient, agentID)
+	})
+	require.NoError(t, err, "could not get logs from Agent ID: %q, err: %s",
+		agentID, err)
+
+	monRegExp := regexp.MustCompile(".*-monitoring$")
+	for i, d := range docs.Hits.Hits {
+		// Lazy way to navigate a map[string]any: convert to JSON then
+		// decode into a struct.
+		jsonData, err := json.Marshal(d.Source)
+		if err != nil {
+			t.Fatalf("could not encode document source as JSON: %s", err)
+		}
+
+		doc := ESDocument{}
+		if err := json.Unmarshal(jsonData, &doc); err != nil {
+			t.Fatalf("could not unmarshal document source: %s", err)
+		}
+
+		if monRegExp.MatchString(doc.Component.ID) {
+			t.Errorf("[%d] Document on index %q with 'component.id': %q "+
+				"and 'elastic_agent.id': %q. 'elastic_agent.id' must not "+
+				"end in '-monitoring'\n",
+				i, d.Index, doc.Component.ID, doc.ElasticAgent.ID)
+		}
+	}
+}
+
+func findESDocs(t *testing.T, findFn func() (estools.Documents, error)) estools.Documents {
+	var docs estools.Documents
+
+	require.Eventually(
+		t,
+		func() bool {
+			var err error
+			docs, err = findFn()
+			return err == nil
+		},
+		3*time.Minute,
+		15*time.Second,
+	)
+
+	// TODO: remove after debugging
+	t.Log("--- debugging: results from ES --- START ---")
+	for _, doc := range docs.Hits.Hits {
+		t.Logf("%#v", doc.Source)
+	}
+	t.Log("--- debugging: results from ES --- END ---")
```

Comment:
Is the TODO still relevant ?

Reply:
It's not my code, but I also believe it's not relevant, I'll remove it. Thanks for catching it!